packages/ra-core/src/core/CoreAdminContext.tsx

```diff
@@ -187,9 +191,11 @@ React-admin requires a valid dataProvider function to work.`);
 
     const finalAuthProvider = useMemo(
         () =>
-            authProvider instanceof Function
-                ? convertLegacyAuthProvider(authProvider)
-                : authProvider,
+            authProvider
+                ? authProvider instanceof Function
+                    ? convertLegacyAuthProvider(authProvider)
+                    : authProvider
+                : defaultAuthProvider,
```

Comment:
Just found out this is a BC, because now all admins have a default non-null authProvider.
So this breaks code like the following:

```tsx
        <Root
            label={
                authProvider != null
                    ? 'ra.auth.user_menu'
                    : 'ra.configurable.customize'
            }
            icon={
                authProvider ? (
                    <Avatar src={identity.avatar} />
                ) : (
                    <SettingsIcon />
                )
            }
            {...props}
        />
```

Is there a way to avoid it?

Reply:
https://github.com/marmelab/react-admin/pull/9861 attempts to fix this BC